examples/skyline_selection_results.py

```diff
@@ -0,0 +1,114 @@
+import numpy as np
+import pandas as pd
+import os, pathlib, itertools
+import warnings
+
+warnings.simplefilter('ignore')
+os.chdir('..')
+
+from fp.traindata_samplers import CompleteData
+from fp.missingvalue_handlers import CompleteCaseAnalysis
+from fp.dataset_experiments import GermanCreditDatasetSexExperiment
+from fp.scalers import NamedStandardScaler
+from fp.learners import LogisticRegression, DecisionTree
+from fp.post_processors import NoPostProcessing, RejectOptionPostProcessing
+from fp.pre_processors import NoPreProcessing, DIRemover
+
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+# creating list of parameters that we will alter to observe variations
+seeds = [0xbeef, 0xcafe, 0xdead, 0xdeadcafe]
+learners = [LogisticRegression()]
+
+processors = [(NoPreProcessing(), NoPostProcessing()), (DIRemover(1.0), NoPostProcessing()), (NoPreProcessing(), RejectOptionPostProcessing())]
+
+
+skyline_order = ['accuracy', 'selection_rate', 'false_discovery_rate']
+skyline_formula = {'accuracy': 0.5, 'selection_rate': 0.3, 'false_discovery_rate': 0.2}
+
+def calculate_metrics(seed, learners, pre_processors, post_processors, skyline_strategy):
+    '''
+        Experiment function to run the experiments with multiple combinations of learners and processors in the input
+    '''
+    exp = GermanCreditDatasetSexExperiment(
+        fixed_random_seed=seed,
+        train_data_sampler=CompleteData(),
+        missing_value_handler=CompleteCaseAnalysis(),
+        numeric_attribute_scaler=NamedStandardScaler(),
+        learners=learners,
+        pre_processors=pre_processors,
+        post_processors=post_processors,
+        optimal_validation_strategy=skyline_strategy)
+    exp.run()
+    return exp.generate_file_path()
+
+def run_exp(seeds, learners, processors, skyline_strategy):
+    '''
+        This is the main driver function that calls the calculate_metrics to give metrices on combinations of various learners, pre and post processing techniques.
+    '''
+    skyline_res_folder = {}
+    for seed in seeds:
+        skyline_res_folder[seed] = calculate_metrics(seed, learners, [x[0] for x in processors], [x[1] for x in processors], skyline_strategy)
```

Comment:
the comprehensions for the preprocessors are difficult to read, please just give two lists to the function

Reply:
Fixed.